File: rest-kotlin/src/main/kotlin/edu/drexel/ws/kotlin/rest/demo/Publication.kt
package edu.drexel.ws.kotlin.rest.demo

import com.fasterxml.jackson.annotation.JsonInclude
import java.util.*

data class RequestError(var id:Int, var msg:String)

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Publication (
        var id: Int,
        var title: String,
        var cite: String,
        var slides: String?,
        var link: String?,
        var url: String?,
        var pubDate: Date?,
        var `abstract`: String?,
        var error: RequestError?)

File: rest-scala/src/main/scala/edu/drexel/ws/scala/rest/demo/Publication.scala
package edu.drexel.ws.scala.rest.demo

case class Publication ( id:      Option[Integer] = None,
                         title:   Option[String] = None,
                         cite:    Option[String] = None,
                         slides:  Option[String]= None,
                         link:    Option[String]= None,
                         url:     Option[String]= None,
                         pubDate: Option[String]= None,
                         `abstract`: Option[String]= None,
                         error:   Option[PublicationError]= None)
